File: app/src/main/java/com/crossbowffs/quotelock/modules/hitokoto/app/HitkotoConfigActivity.java
package com.crossbowffs.quotelock.modules.hitokoto.app;

import static com.crossbowffs.quotelock.modules.hitokoto.consts.HitokotoPrefKeys.PREF_HITOKOTO;
import static com.crossbowffs.quotelock.modules.hitokoto.consts.HitokotoPrefKeys.PREF_HITOKOTO_TYPE_INT;
import static com.crossbowffs.quotelock.modules.hitokoto.consts.HitokotoPrefKeys.PREF_HITOKOTO_TYPE_STRING;

import android.app.Activity;
import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.RadioButton;
import android.widget.RadioGroup;

import com.crossbowffs.quotelock.R;

public class HitkotoConfigActivity extends Activity implements RadioGroup.OnCheckedChangeListener {
    private SharedPreferences mHitokotoPreferences;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mHitokotoPreferences = getSharedPreferences(PREF_HITOKOTO, Context.MODE_PRIVATE);
        setContentView(R.layout.radio_hitokoto_quote);

        RadioGroup radioGroup = (RadioGroup)findViewById(R.id.module_hitokoto_activity_radiogroup);
        int queryValueIndex = mHitokotoPreferences.getInt(PREF_HITOKOTO_TYPE_INT, 0);
        RadioButton button = (RadioButton)radioGroup.getChildAt(queryValueIndex);
        button.setChecked(true);

        radioGroup.setOnCheckedChangeListener(this);
    }

    @Override
    public void onCheckedChanged(RadioGroup group, int checkedId) {
        String[] queryValueArray = getResources().getStringArray(R.array.hitokoto_type_values);
        int queryValueIndex = group.indexOfChild(findViewById(checkedId));
        String queryValue = queryValueArray[queryValueIndex];

        mHitokotoPreferences.edit()
            .putInt(PREF_HITOKOTO_TYPE_INT, queryValueIndex)
            .putString(PREF_HITOKOTO_TYPE_STRING, queryValue)
            .apply();
    }
}

File: app/src/main/java/com/crossbowffs/quotelock/modules/hitokoto/app/HitkotoConfigActivity.kt
package com.crossbowffs.quotelock.modules.hitokoto.app

import android.app.Activity
import android.os.Bundle
import android.view.View
import android.widget.RadioButton
import android.widget.RadioGroup
import com.crossbowffs.quotelock.modules.hitokoto.consts.HitokotoPrefKeys
import com.crossbowffs.quotelock.modules.hitokoto.hitokotoDataStore
import com.google.android.material.appbar.MaterialToolbar
import com.yubyf.quotelockx.R
import kotlinx.coroutines.runBlocking

class HitkotoConfigActivity : Activity(), RadioGroup.OnCheckedChangeListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.radio_hitokoto_quote)

        // Toolbar
        findViewById<MaterialToolbar>(R.id.toolbar).setNavigationOnClickListener { onBackPressed() }

        val radioGroup = findViewById<View>(R.id.module_hitokoto_activity_radiogroup) as RadioGroup
        val queryValueIndex = runBlocking {
            hitokotoDataStore.getIntSuspend(HitokotoPrefKeys.PREF_HITOKOTO_TYPE_INT,
                0)
        }
        val button = radioGroup.getChildAt(queryValueIndex) as RadioButton
        button.isChecked = true
        radioGroup.setOnCheckedChangeListener(this)
    }

    override fun onCheckedChanged(group: RadioGroup, checkedId: Int) {
        val queryValueArray = resources.getStringArray(R.array.hitokoto_type_values)
        val queryValueIndex = group.indexOfChild(findViewById(checkedId))
        val queryValue = queryValueArray[queryValueIndex]
        hitokotoDataStore.run {
            put(HitokotoPrefKeys.PREF_HITOKOTO_TYPE_INT, queryValueIndex)
            put(HitokotoPrefKeys.PREF_HITOKOTO_TYPE_STRING, queryValue)
        }
    }
}